crates/stages/src/stages/merkle.rs

```diff
@@ -5,28 +5,44 @@ use reth_db::database::Database;
 use std::fmt::Debug;
 use tracing::*;
 
-const MERKLE_EXECUTION: StageId = StageId("MerkleExecuteStage");
-const MERKLE_UNWIND: StageId = StageId("MerkleUnwindStage");
+/// The [`StageId`] of the merkle hashing execution stage.
+pub const MERKLE_EXECUTION: StageId = StageId("MerkleExecuteStage");
 
-/// Merkle stage uses input from [AccountHashingStage] and [StorageHashingStage] stages
-/// and calculated intermediate hashed and state root.
-/// This stage depends on the Account and Storage stages. It will be executed after them during
-/// execution, and before them during unwinding.
+/// The [`StageId`] of the merkle hashing unwind stage.
+pub const MERKLE_UNWIND: StageId = StageId("MerkleUnwindStage");
+
+/// The merkle hashing stage uses input from
+/// [`AccountHashingStage`][crate::stages::AccountHashingStage] and
+/// [`StorageHashingStage`][crate::stages::AccountHashingStage] to calculate intermediate hashes
+/// and state roots.
+///
+/// This stage should be run with the above two stages, otherwise it is a no-op.
+///
+/// This stage is split in two: one for calculating hashes and one for unwinding. TODO: Why?
+/// The order of these two variants is important. The unwind variant should be added to the pipeline
+/// before the execution variant.
+///
+/// An example pipeline to only hash state would be:
+///
+/// - [`MerkleStage::Unwind`]
+/// - [`AccountHashingStage`][crate::stages::AccountHashingStage]
+/// - [`StorageHashingStage`][crate::stages::StorageHashingStage]
+/// - [`MerkleStage::Execution`]
 #[derive(Debug)]
-pub struct MerkleStage {
-    /// Flag if true would do `execute` but skip unwind but if it false it would skip execution but
-    /// do unwind.
-    pub is_execute: bool,
+pub enum MerkleStage {
```

Comment:
I opted to refactor this stage because `is_execute: bool` was not super descriptive.